src/main/java/jnr/ffi/LibraryOption.java

```diff
@@ -51,15 +51,33 @@
      * A function mapper which maps from java function names to native function names.
      */
     FunctionMapper,
-    
+
     /**
      * The type of calling convention.
+     *
      * @see CallingConvention
      */
     CallingConvention,
 
     /**
      * Load the library into memory immediately, instead of lazily loading it
      */
-    LoadNow
+    LoadNow,
+
+    /**
+     * Relevant for GNU/Linux {@link Platform.Linux} only
+     *
+     * Prefer custom paths over system paths when loading a library, even if the custom path has a lower version.
+     *
+     * By default JNR-FFI will choose the library of the desired name with the highest version, whether in the custom
+     * paths or the system default paths.
+     *
+     * This can be a problem if you are distributing your own library for example {@code libfoo.so} and the system
+     * paths <i>happen</i> to have a {@code libfoo.so.4} for example, in which case JNR-FFI will prefer the
+     * higher version despite your explicit custom paths.
+     *
+     * By using this option, JNR-FFI will know to prefer the custom paths, even if they have a lower version, this
+     * ensures consistent behaviors across environments.
+     */
+    PreferCustomPaths
```

Comment:
Rather than `PreferCustomPaths`, maybe `RespectLibraryPath` sounds interesting.
Should the pre-existing option be a default and included here now: `PreferLatestVersion`

Reply:
@fzakaria I want to add as few options as possible, also we should keep the current behavior as the default for backwards compatibility even if I personally don't like it, I don't want upgrading users to get confused. Hence my decision for the new option to be the one that isn't the current default. What do you think about the name of the option @headius ?

Reply:
I generally advocate that all possible options should be enumerated, so that it is possible to force the default behavior if for some reason that default is no longer used (embedded use, custom build, future behavior change).

**However** this is not the approach that has been taken thus far, for example the `LoadNow` option does not have an equivalent `LazyLoad` option. I think we can skip adding the default `PreferLatestVersion` option until we revisit how options should be used.

Reply:
It is also trivial and non-breaking for us to add such an option in the future, if it turns out to be needed by some user.

Reply:
@headius My exact thoughts, you said it better than I could! 😁